src/stack.ts

```diff
@@ -176,8 +176,13 @@ export function stack(m: Mark | MarkDef, encoding: Encoding<string>): StackPrope
   if (encoding[dimensionChannel]) {
     const dimensionDef = encoding[dimensionChannel];
     const dimensionField = isFieldDef(dimensionDef) ? vgField(dimensionDef, {}) : undefined;
+    const hasSameDimensionAndStackedField = dimensionField && dimensionField === stackedField;
```

Comment:
just having the same input field doesn't mean that it's the same transformed field? I think we need to check if they have the same aggregate/bin/timeUnit?

Reply:
Ah interesting. Do you know if we have helper utilities for checking if two encodings are "equivalent"? (I.e. matching in field name and any relevant inline transformations like bin/timeUnit/bin? 

I looked in channelDef just now and didn't see anything that matches. I can add one if it seems like it would be appropriate here.

Reply:
Reading this method, I think we check if it's the same field by converting a field def to vgField first. 

For example, see this block in this method. 

```ts
const dimensionField = isFieldDef(dimensionDef) ? vgField(dimensionDef, {}) : undefined;
if (dimensionField && dimensionField !== stackedField) {
```